pkg/sse/sse.go

```diff
@@ -26,17 +27,25 @@ type Writer struct {
 func (w *Writer) Write(p []byte) (int, error) {
 	w.Lock()
 	defer w.Unlock()
-
-	if _, err := w.Writer.Write([]byte("data: ")); err != nil {
-		return 0, err
-	}
-	if _, err := w.Writer.Write(p); err != nil {
-		return 0, err
+	for _, line := range bytes.Split(p, []byte("\n")) {
+		data := fmt.Sprintf("data: %s\n", line)
+		if _, err := w.Writer.Write([]byte(data)); err != nil {
```

Comment:
Should we worry about the return value for bytes written?  (not that the current code does either)